beancount_import/matching.py

```diff
@@ -255,6 +283,81 @@ def add_posting(self, entry: Transaction, mp: MatchablePosting):
         group = self._postings.setdefault(_date_key(entry, mp), {})
         group[source_posting_ids] = (entry, mp)
 
+        weight = get_posting_weight(mp.posting)
+        if weight is None:
+            return
+
+        for dc in self.fuzz_date_currency_key(self._date_currency_key(entry, mp)):
+            bisect.insort(
+                self._date_currency[dc],
+                self._search_posting(entry, mp))
+
+    def search_postings(self,
+                        entry: Transaction,
+                        mps: List[MatchablePosting],
+                        cmp_callable: Callable,
+                        negate=False) -> Iterable[SearchPosting]:
+        # Prepare postings for searching
+        postings_date_currency = collections.defaultdict(list)
+        
+        for mp in mps:
+            weight = get_posting_weight(mp.posting)
+            if weight is None:
+                continue
+            if negate:
+                weight = -weight
+            bisect.insort(
```

Comment:
Definitely should avoid using `bisect.insort` here (instead could sort all of the lists at the end), but if you change to just looking up each posting match individually, there is no need for this data structure at all.

Possibly with this issue fixed you could increase the limit on number of matching postings to allow, since as it is currently, it is O(n**2), where n is the number of matchable postings with a given (date, currency).